### ISSUE TITLE
Finetuning with Muon

### ISSUE THREAD

**Issue by dxqbYD** [KellerJordan/Muon#25]
Thanks for your work first of all!

You have posed this question in your write-up:
- Is it possible that Muon works only for pretraining, and won’t work for finetuning or reinforcement learning workloads?

Here are some anecdotal experiments using LoRA finetuning of Flux Dev1, an image generation model that is notoriously difficult to finetune on multiple concepts/tasks. The reasons for this are unknown, but believed to be related to the fact that this model was not trained on raw data but distilled from a teacher model.

These are validation loss graphs
![Image](https://github.com/user-attachments/assets/632f0e04-e768-4707-b182-d5c44a8fa17b)
![Image](https://github.com/user-attachments/assets/c6bf98cc-22f3-436b-87f5-eaa9397c4dde)
![Image](https://github.com/user-attachments/assets/f21cacfe-7e31-4b1f-bd64-7be9007cbb6e)
![Image](https://github.com/user-attachments/assets/6b655419-ef6b-41d4-a2ab-f9b117eb9ec7)
![Image](https://github.com/user-attachments/assets/60ac146c-d0bb-4683-94c8-c25290960b00)

The first one is the average, the other ones are the 4 different tasks that are trained. Green is Muon, Orange is AdamW. You can tell that while AdamW learns well for a while, it seems to hit a wall at some point where the tasks probably compete against each other and the validation loss becomes unstable. Muon seems to deal with this much better.

In the last graph in white, you can see for reference a single-task training of the same task using AdamW. It can easily get where Muon goes, but only if there are no competing tasks. 
[The white line is a bit unstable because with 1 task it's only 20 training samples and it overfits fast. For the same reason, the validation loss goes back up after a minimum]

The above trainings were with batch size 4 on 4 tasks. Here is the same 4 tasks but batch size only 1:

![Image](https://github.com/user-attachments/assets/980a30a8-6361-436c-af13-27299bb65305)
Muon clearly outperforms AdamW again because of the competing tasks.

Feel free to close this issue if this is not the right place, but I thought this might be an interesting anecdote for you, because it indicates that Muon is not only faster to the goal, but can find better parameters for a difficult problem.

the AdamW hyperparameters I have used, I have used and tuned many times - Muon for the first time. So if there is any hyperparameter bias, it is probably in favor of AdamW.

**Comment by davnords** [comment]
Good work! What is the learning rate for muon here?

**Comment by dxqbYD** [comment]
it's been a while but I've found this in the filenames of the samples: muon-0.003-mom0.95-decay0.01
note that this was a LoRA with Alpha 1.0
